src/impala/sema/infersema.cpp

```diff
@@ -135,6 +135,9 @@ class InferSema : public TypeTable {
             return t->rebuild(nargs);
         }
 
+        if (u->isa<NoRetType>()) return unify(representative(t), representative(u))->type;
+        if (t->isa<NoRetType>()) return unify(representative(u), representative(t))->type;
+
```

Comment:
I don't think this is a good idea:
Say, ```t``` is ```noret ``` and ```u``` is ```int```. Then, this code will unify ```t``` and ```u``` - but I don't think those type are supposed to be the same thing.